### DISCUSSION TITLE
One individual cannot handle the continued development of this repo

### DISCUSSION THREAD

**Discussion by 34j** [voicepaw/so-vits-svc-fork#303] [General]
- Lack of an initial model: cannot train QuickVC at all with an A4000 or so; tried half-precision training on T4, but not fast enough at all; No money to rent a better GPU; (Looking for initial model donors.)
- TPU support was too hard and IPU support is doomed; There are no resources to experiment with reducing the size of the model
- New models (such as 5.0 or RVC) are released day by day and implementation is not in time. I am looking for contributors, but I'm not even sure what the point is to implement them after the fact.
- Real-time support is no longer novel since many repositories have already referenced our repo code and implemented it.
- Is there anyone who wants to be a maintainer of this repository in this terrible situation?

**Comment by sbersier** [comment]
1) Regarding the training:
Have you considered lowering the sampling rate from 44100 to 22050? Wouldn't it help?
To be perfectly honnest, I don't fully understand this choice (i.e. 44k) since human voice extends mostly up to 10k Hz. So, there is not much to gain by have a sampling rate above 22k, is there? And, in fact, I think that 16k Hz would already be quite good.  

By doing this you would be able to fit twice as much data in VRAM.

In fact, I'm wondering if the result couldn't be even better since sampling frequency regions that don't contain much useful information might be conterproductive. Also setting mel_fmax to 11025 Hz would increase the spectral resolution. You might even be able to decrease n_mel_channels (if it helps training). 


2) Could collaborative training be a solution? 

Like in: https://huggingface.co/blog/collaborative-training

In fact, as I understand it, it is based on hivemind:
https://learning-at-home.github.io/

If you can put in place such a solution, I would be happy to give GPU time.
In this example, they trained a model for bengali language with 40 volonteers (each one having different hardware and bandwidth). I highly recommend you have a look at it if you haven't already had.


Anyway, I hope you call for help will be heard. 

Best regards

**Reply by ADR3-N** [reply]
I would be very interested to see results if you are able to get something running on 22k, since I have a very slow machine. If you do, or anyone else does, please @ me on twitter, OGAdrean. I really only need 22k files, and this fork is working great for me, but I imagine a slightly lower sample rate would not only accelerate training but allow me to increase batch size.

**Reply by GarrettConway** [reply]
Just some technical notes for anyone interested:
 You should be able to adjust the sampling rate in the config and train on it. Admittedly, I haven't devoted any time to personally tinkering with this part recently, but I don't believe any changes would have broken that functionality. (If someone runs into any problems, feel free to @ me and I can fix them). "svc pre-resample --sampling-rate" will also let you set that in the preprocessing.
Sidenote: The contentvec model is always fed audio sampled at 16k since that's what their pretrained model uses - the sampling rate for the svc model mostly affects the vocoder and the mel loss.

Cheers,
Garrett

**Comment by Air1N** [comment]
SVC 5.0 is already out?

**Comment by Z3Coder** [comment]
How long would it take to train initial models on a RTX3090? 
Is dataset available for the models and if yes, how big is it?
Depending on the answers, maybe i could train the models, or at least one.
Thank you for your work! @34j

**Comment by SamueleLorefice** [comment]
What is needed for QuickVC training? Provided we have a dataset available I could go on and do initial training on my 4090 (or rent multiple 4090s for some hours).

I'd have to look deeper into this to try and maintain the repo, but lending an hand in my (albeit limited) free time is something I could do. Not alone certainly. But I'm up for discussion. And looks like I'm not alone here either.

**Reply by 34j** [reply]
Migrate to #579

**Comment by Onako2** [comment]
> * Lack of an initial model: cannot train QuickVC at all with an A4000 or so; tried half-precision training on T4, but not fast enough at all; No money to rent a better GPU; (Looking for initial model donors.)
> 
>     * TPU support was too hard and IPU support is doomed; There are no resources to experiment with reducing the size of the model
> 
>     * New models (such as 5.0 or RVC) are released day by day and implementation is not in time. I am looking for contributors, but I'm not even sure what the point is to implement them after the fact.
> 
>     * Real-time support is no longer novel since many repositories have already referenced our repo code and implemented it.
> 
>     * Is there anyone who wants to be a maintainer of this repository in this terrible situation?

I would like to help but I don't know much programming. I am still in school and will stay for a long time and will complete school in 6 years :(
